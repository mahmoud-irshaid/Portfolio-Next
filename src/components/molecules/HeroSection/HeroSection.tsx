import { TypeWriter } from '@components/atoms/TypeWriter';

import Image from 'next/image';
import Link from 'next/link';

export const HeroSection = () => {
	const talkAbout = [
		'React',
		'React Native',
		'JavaScript',
		'TypeScript',
		'Redux',
		'Next.js',
		'RESTful APIs',
		'HTML',
		'CSS',
		'Expo',
		'Jest',
		'Node',
		'Express',
	];

	return (
		<section id="hero" className="relative isolate overflow-hidden pt-14">
			<div className="bg-gradient-from-t absolute inset-x-0 bottom-0 -z-10 h-24 from-off-white dark:from-black sm:h-32" />
			<div
				className="absolute inset-y-0 right-1/2 -z-10 -mr-96 w-[200%] origin-top-right skew-x-[-30deg] dark:shadow-slate-700 sm:-mr-80 lg:-mr-96 lg:shadow-xl"
				aria-hidden="true"
			/>
			<div className="mx-auto max-w-7xl px-6 lg:px-8 lg:py-32">
				<div className="mx-auto max-w-2xl lg:mx-0 lg:grid lg:max-w-none lg:grid-cols-2 lg:gap-x-16 lg:gap-y-8 xl:grid-cols-1 xl:grid-rows-1 xl:gap-x-8">
					<h1 className="headline max-w-4xl text-4xl font-bold tracking-tight sm:text-6xl lg:col-span-2 xl:col-auto">
						Hey, I&apos;m Mahmoud Irshaid
					</h1>
					<div className="mt-6 max-w-xl lg:mt-0 xl:col-end-1 xl:row-start-1">
						<p className="text-lg leading-8 text-slate-600 dark:text-off-white">
							I am a passionate Software Engineer, specialised in front-end
							development using React/React-Native and TypeScript. I create amazing web and mobile applications to make the internet a better
							place.
							<br />
							You can talk to me about <TypeWriter strings={talkAbout} />.
						</p>
						<div className="mt-10 flex items-center gap-x-6">
							<Link
								href="/contact"
								className="rounded-md bg-blue-700 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm transition-all hover:bg-blue-600 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-600"
							>
								Get in touch
							</Link>
							<Link
								href="/about"
								className="text-sm font-semibold leading-6 text-slate-900 transition-all hover:text-slate-600 dark:text-slate-400 dark:hover:text-slate-200"
							>
								Read more <span aria-hidden="true">â†’</span>
							</Link>
						</div>
					</div>
					<Image
						src="/assets/mahmoud.jpg"
						alt="Photo of Mahmoud Irshaid"
						height={300}
						width={450}
						loading="eager"
						className="mt-10 aspect-[6/5] max-w-full rounded-2xl object-cover sm:mt-16 lg:ml-20 lg:mt-0 lg:max-w-none xl:row-span-2 xl:row-end-2 xl:mt-36"
					/>
				</div>
			</div>
			<div className="absolute inset-x-0 bottom-0 -z-10 h-24 bg-gradient-to-t from-off-white dark:from-black sm:h-32" />
		</section>
	);
};
