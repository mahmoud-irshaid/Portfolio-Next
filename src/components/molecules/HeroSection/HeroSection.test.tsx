import { HeroSection } from './HeroSection';

import { render } from '@testing-library/react';

describe('HeroSection', () => {
	it('renders correctly', () => {
		const { container } = render(<HeroSection />);
		expect(container).toMatchSnapshot();
	});
});
