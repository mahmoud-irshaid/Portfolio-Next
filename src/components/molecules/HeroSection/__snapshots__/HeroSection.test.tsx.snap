// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection renders correctly 1`] = `
<div>
  <section
    class="relative isolate overflow-hidden pt-14"
    id="hero"
  >
    <div
      class="bg-gradient-from-t absolute inset-x-0 bottom-0 -z-10 h-24 from-off-white dark:from-black sm:h-32"
    />
    <div
      aria-hidden="true"
      class="absolute inset-y-0 right-1/2 -z-10 -mr-96 w-[200%] origin-top-right skew-x-[-30deg] dark:shadow-slate-700 sm:-mr-80 lg:-mr-96 lg:shadow-xl"
    />
    <div
      class="mx-auto max-w-7xl px-6 lg:px-8 lg:py-32"
    >
      <div
        class="mx-auto max-w-2xl lg:mx-0 lg:grid lg:max-w-none lg:grid-cols-2 lg:gap-x-16 lg:gap-y-6 xl:grid-cols-1 xl:grid-rows-1 xl:gap-x-8"
      >
        <h1
          class="headline max-w-2xl text-4xl font-bold tracking-tight sm:text-6xl lg:col-span-2 xl:col-auto"
        >
          Hey, I'm Mahmoud Irshaid
        </h1>
        <div
          class="mt-6 max-w-xl lg:mt-0 xl:col-end-1 xl:row-start-1"
        >
          <p
            class="text-lg leading-8 text-slate-600 dark:text-off-white"
          >
            I am a passionate Software Engineer, specialised in front-end development using React and TypeScript. As an advocate for web performance and accessibility and an evangelist for the Jamstack, I create amazing web applications to make the internet a better place.
            <br />
            You can talk to me about 
            <span>
              <span />
              <span
                class="typed-cursor"
              >
                |
              </span>
            </span>
            .
          </p>
          <p
            class="text-lg leading-8 text-slate-600 dark:text-off-white"
          >
            P.S. this website is open-source and available on
             
            <a
              class="underlined relative border-b-2 border-grey-300 font-bold hover:border-b-0 dark:border-grey-700"
              href="https://github.com/jakeherp/portfolio"
              rel="noopener noreferrer"
              target="_blank"
              title="Link to Github repository"
            >
              Github
            </a>
            .
          </p>
          <div
            class="mt-10 flex items-center gap-x-6"
          >
            <a
              class="rounded-md bg-blue-700 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm transition-all hover:bg-blue-600 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-600"
              href="/contact"
            >
              Get in touch
            </a>
            <a
              class="text-sm font-semibold leading-6 text-slate-900 transition-all hover:text-slate-600 dark:text-slate-400 dark:hover:text-slate-200"
              href="/about"
            >
              Read more 
              <span
                aria-hidden="true"
              >
                â†’
              </span>
            </a>
          </div>
        </div>
        <img
          alt="Photo of Mahmoud Irshaid"
          class="mt-10 aspect-[6/5] max-w-full rounded-2xl object-cover sm:mt-16 lg:ml-20 lg:mt-0 lg:max-w-none xl:row-span-2 xl:row-end-2 xl:mt-36"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="eager"
          src="/_next/image?url=%2Fassets%2Fmahmoud.jpg&w=1080&q=75"
          srcset="/_next/image?url=%2Fassets%2Fmahmoud.jpg&w=640&q=75 1x, /_next/image?url=%2Fassets%2Fmahmoud.jpg&w=1080&q=75 2x"
          style="color: transparent;"
          width="450"
        />
      </div>
    </div>
    <div
      class="absolute inset-x-0 bottom-0 -z-10 h-24 bg-gradient-to-t from-off-white dark:from-black sm:h-32"
    />
  </section>
</div>
`;
